models/classes/eventHandler/ResultTransmissionEventHandler/AsynchronousResultTransmissionEventHandler.php

```diff
@@ -66,6 +68,13 @@ private function packVariables(array $variables): array
     {
         return array_map(static function ($record) {
             if (is_object($record)) {
+                if ($record instanceof ResponseVariable) {
+                    $value = $record->getValue();
+                    if ($value instanceof QtiFlysystemFile) {
```

Comment:
Perhaps, relying on the service locator interface would be a more generic solution, WDYT?
```suggestion
                    if ($value instanceof \Laminas\ServiceManager\ServiceLocatorAwareInterface) {
```

Reply:
the problem is this [place](https://github.com/oat-sa/generis/blob/master/common/oatbox/filesystem/FileSystemHandler.php#L86) 
for application instances with configured `s3` `QtiFlysystemFile` contain `S3Client` which [don't want to be serialized](https://github.com/aws/aws-sdk-php/blob/3.173.19/src/AwsClient.php#L275)

Reply:
maybe i describe problem in commits messages and summary not enough explicitly

Reply:
Right, my mistake.
Thanks for the clarification!

Reply:
Wouldn't it be better to utilize `__sleep()` `__wakeup()` on `FileSystemHandler` itself, so that it doesn't even try serializing the filesystem property?

Reply:
Hm, OK, never mind. I don't think this will work with inheritance properly.

Reply:
just check it `filesystem` is accessable in `QtiFlysystemFile`

Reply:
WDYT https://github.com/oat-sa/generis/pull/979

Reply:
Nice, I didn't notice the parent class defines its properties' visibility as `protected`!